packages/vanilla/src/controls/input.control.tsx

```diff
@@ -1,34 +1,59 @@
 import * as React from 'react';
 import {
   computeLabel,
+  Control,
+  ControlElement,
   ControlProps,
+  ControlState,
   DispatchField,
   formatErrorMessage,
   isControl,
+  isDescriptionHidden,
+  JsonForms,
   mapStateToControlProps,
   RankedTester,
   rankWith,
-  registerStartupRenderer
+  registerStartupRenderer,
+  resolveSchema
 } from '@jsonforms/core';
 import { connect } from 'react-redux';
 
-export const InputControl =
-  ({ classNames, id, errors, label, uischema, schema, visible, required }: ControlProps) => {
-  const isValid = errors.length === 0;
-  const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+export class InputControl extends Control<ControlProps, ControlState> {
+  render() {
+    const { classNames, id, errors, label, uischema, schema, visible, required } = this.props;
+    const isValid = errors.length === 0;
+    const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+    const controlElement = uischema as ControlElement;
+    const resolvedSchema = resolveSchema(schema, controlElement.scope.$ref);
+    const inputDescriptionClassName =
+      JsonForms.stylingRegistry.getAsClassName('input-description');
+    const description = resolvedSchema.description === undefined ? '' : resolvedSchema.description;
 
-  return (
-    <div className={classNames.wrapper} hidden={!visible}>
-      <label htmlFor={id} className={classNames.label}>
-        {computeLabel(label, required)}
-      </label>
-      <DispatchField uischema={uischema} schema={schema}/>
-      <div className={divClassNames}>
-        {!isValid ? formatErrorMessage(errors) : ''}
+    return (
+      <div
+        className={classNames.wrapper}
+        hidden={!visible}
+        onFocus={() => this.onFocus()}
+        onBlur={() => this.onBlur()}
+      >
+        <label htmlFor={id} className={classNames.label}>
+          {computeLabel(label, required)}
+        </label>
+        <DispatchField uischema={uischema} schema={schema}/>
+        <div className={divClassNames}>
```

Comment:
can we unite the two divs as you did in the material renderer?
Maybe make the styling also less intrusive?

Reply:
updated